layers/+spacemacs/spacemacs-org/local/space-doc/space-doc.el

```diff
@@ -102,11 +106,69 @@ It is set by `spacemacs//space-doc-set-space-doc-cache'.")
 
 (defun spacemacs//space-doc-hide-line-numbers (&optional enable)
   "If ENABLE is non-nil then toggle off the line numbers."
+  ;; FIXME: Can we prevent it from producing a message?
   (if enable
       (spacemacs/toggle-line-numbers-off)
     (when dotspacemacs-line-numbers
       (spacemacs/toggle-line-numbers-on))))
 
+(defvar spacemacs--space-doc-this-window-name nil)
+
+(defun spacemacs//space-doc-on-window-width-changed-hook ()
+  "Used in `spacemacs//space-doc-modf-center-buffer'"
+  (setq spacemacs--space-doc-this-window-name
+        (get-buffer-window))
+  (run-with-idle-timer 0 nil  (lambda ()
+                                (with-selected-window
+                                    spacemacs--space-doc-this-window-name
+                                  (delete-other-windows)))))
+
+(defun spacemacs//space-doc-calc-fringe-width ()
+  "Calculates fringe width for `spacemacs//space-doc-modf-center-buffer'"
+  (-(/ (- (frame-pixel-width)
+          (car(window-text-pixel-size)))
+       ;; 50 * 2 extra pixels for the deferred shenanigans.
+       2) 50))
+
+(defun spacemacs//space-doc-modf-center-buffer (&optional enable)
+  "If ENABLE is non-nil then toggle on buffer centering."
+
+  (if enable (let ((fringe-width (spacemacs//space-doc-calc-fringe-width)))
+               (set (make-local-variable 'spacemacs--space-doc-origin-left-fringe-width)
+                    (or left-fringe-width 0))
+               (set (make-local-variable 'spacemacs--space-doc-origin-right-fringe-width)
+                    (or right-fringe-width 0))
+               (set (make-local-variable 'spacemacs--space-doc-window-old-width)
+                    (window-width))
+               (when (> fringe-width 0)
+                 (delete-other-windows)
+                 (setq left-fringe-width fringe-width
+                       right-fringe-width fringe-width)
+                 (setq-local org-startup-folded nil))
+               (add-hook 'window-configuration-change-hook
+                         'spacemacs//space-doc-on-window-width-changed-hook
+                         nil t))
+
+    (setq  left-fringe-width spacemacs--space-doc-origin-left-fringe-width
+           right-fringe-width spacemacs--space-doc-origin-right-fringe-width)
+    (remove-hook 'window-configuration-change-hook
+                 'spacemacs//space-doc-on-window-width-changed-hook
+                 t)
+    (kill-local-variable org-startup-folded))
+
+  ;; FIXME: Find a better way to update fringes.
+  (let (this-buffer-name (buffer-name))
```

Comment:
Did you try `(set-window-buffer nil (current-buffer))`? The documentation for `left-fringe-width` says:
> To make the change take effect, call `set-window-buffer'.